query_optimizer/ExecutionGenerator.cpp

```diff
@@ -679,13 +688,72 @@ void ExecutionGenerator::convertFilterJoin(const P::FilterJoinPtr &physical_plan
       std::piecewise_construct,
       std::forward_as_tuple(physical_plan),
       std::forward_as_tuple(probe_relation_info->producer_operator_index,
-                            probe_relation_info->relation));
+                            probe_relation_info->relation,
+                            probe_relation_info->output_destination_index));
 
   DCHECK(lip_filter_generator_ != nullptr);
   lip_filter_generator_->addFilterJoinInfo(physical_plan,
                                            build_filter_operator_index);
 }
 
+namespace {
+
+bool areSamePartitionSchemeHeaders(const PartitionSchemeHeader &lhs_partition_header,
```

Comment:
I had a discussion with @jianqiao on this issue. He thinks that the partitioning related code should be moved to different functions at least. That way the overall code is easier to read and is also extensible. He's fine with the anonymous namespace arrangement. 

An example is all the new code in ``convertHashJoin`` can be moved to separate function(s) and then called from ``convertHashJoin``.